documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -263,130 +216,99 @@ public void run() {
+application reasons about data from exactly one machine. We can build other applications that
+use a Reader Group configured to read from all of the Streams. To keep it simple, in the sample application the Reader Group only reads from one Stream.
 
-Note that in other cases, if the developer knows the name of the ReaderGroup to
-use and knows it has already been created, s/he can use getReaderGroup() on
-ReaderGroupManager to retrieve the ReaderGroup object by name.
+We can call `createReaderGroup()` with the same parameters multiple times and the same Reader Group will be returned each time after it is initially created (idempotent operation). Note that in other cases, if the developer knows the name of the Reader Group to use and knows it has already been created, they can use `getReaderGroup()` on `ReaderGroupManager` to retrieve the `ReaderGroup` object by name.
 
-So at this point in the code, we have the Scope and Stream set up, we have the
-ReaderGroup created and now we need to create a Reader and start reading Events.
+At this point, we have the Scope and Stream is set up and the `ReaderGroup` object created. Next, we need to create a Reader and start reading Events.
 
 ## Reading Event using an EventStreamReader
 
-Lines 17-36 show an example of setting up an EventStreamReader and reading
-Events using that EventStreamReader.
-
-First, we create a ClientFactory on line 17, in the same way we did it in the
-HelloWorldWriter app.  
-
-Then we use the ClientFactory to create an EventStreamReader object.  There are
-four things the developer needs to create a Reader: a name for the reader, the
-readerGroup it should be part of, the type of object expected on the Stream, the
-serializer to use to convert from the bytes stored in Pravega into the Event
-objects and a ReaderConfig.  Lines 18-21 show the creation of an
-EventStreamReader.  The name of the Reader can be any valid Pravega name
-(numbers and letters).  Of course, the name of the reader is namespaced within
-the Scope.  We talked about the creation of the ReaderGroup in the previous
-section.  Just like with the EventStreamWriter, EventStreamReader uses Java
-generic types to allow a developer to specify a type safe Reader.  In our case,
+First, we create a `ClientFactory` object, the same way we did it in the
+`HelloWorldWriter` application. Then we use the `ClientFactory` to create an `EventStreamReader` object. The following are the four parameters to create a Reader:
+
+ - Name for the Reader.
+ - Reader Group it should be part of.
+ - The type of object expected on the Stream.
+ - Serializer to convert from the bytes stored in Pravega into the Event
+objects and a `ReaderConfig`.
+
+```Java
+EventStreamReader<String> reader = clientFactory.createReader("reader",
+                                                              readerGroup,
+                                                              new JavaSerializer<String>(),
```

Comment:
See my comment above.

Reply:
@Tom @shrids 
The following is added in the content ` **Note:**  Pravega allows users to write their own serializer. For more information and example, please refer to [Pravega Serializer](https://github.com/pravega/pravega-serializer). `

Reply:
@tkaitchuck @shrids 
Issue created : https://github.com/pravega/pravega-serializer/issues/5

Reply:
Accepted Suggestion